src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -24,50 +28,86 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.util")
-)
+	log = golog.LoggerFor("flashlight.proxied")
 
-// HTTPFetcher is a simple interface for types that are able to fetch data over HTTP.
-type HTTPFetcher interface {
-	Do(*http.Request) (*http.Response, error)
-}
+	proxyAddrMutex sync.RWMutex
+	proxyAddr      = eventual.DefaultUnsetGetter()
+
+	// ErrChainedProxyUnavailable indicates that we weren't able to find a chained
+	// proxy.
+	ErrChainedProxyUnavailable = errors.New("chained proxy unavailable")
+)
 
 func success(resp *http.Response) bool {
 	return resp.StatusCode > 199 && resp.StatusCode < 400
 }
 
-// NewChainedAndFronted creates a new struct for accessing resources using chained
-// and direct fronted servers in parallel.
-func NewChainedAndFronted(proxyAddrFN eventual.Getter) *chainedAndFronted {
-	cf := &chainedAndFronted{proxyAddrFN: proxyAddrFN}
+// SetProxyAddr sets the eventual.Getter that's used to determine the proxy's
+// address. This MUST be called before attempting to use the proxied package.
+func SetProxyAddr(addr eventual.Getter) {
```

Comment:
This decision is polemical. Globals vs. argument passing percolating into different layers are at the opposite of the spectrum, which clear differences. But not so clear which one is best.
Obviously globals introduce convenience, but generally reduce testability and encapsulation.
I guess my main question here would be: is this 100% the case where we want to set this only once and is never going to change throughout the life of the program? If so, it should be ok, including for testing.